nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -1119,12 +1119,15 @@ public Description matchVariable(VariableTree tree, VisitorState state) {
       return Description.NO_MATCH;
     }
     VarSymbol symbol = ASTHelpers.getSymbol(tree);
+
     if (symbol.type.isPrimitive() && tree.getInitializer() != null) {
       return doUnboxingCheck(state, tree.getInitializer());
     }
-    if (!symbol.getKind().equals(ElementKind.FIELD)) {
+    if (!symbol.getKind().equals(ElementKind.FIELD)
+        && !handler.isMethodInvocationForOptionalGet(tree.getInitializer(), state.getTypes())) {
```

Comment:
So, not sure if we discussed this already (sorry for taking so long to review this PR), but `isMethodInvocationForOptionalGet` doesn't sound like it should be a `Handler` method. 

Basically, the point of a handler is to abstract away logic the core doesn't need to be aware of, by providing generic plug-in points into the checker workflow. But `isMethodInvocationForOptionalGet` is not generic. It's a check that only makes sense because, here, the core needs to be aware of the optional get checking (which is what the shadow field approach would avoid).

If we are going to need a hack where the core needs to check that the method matches `Optional.get` and equivalent, that should be an static method in a utility class called from both here and `OptionalEmptinessHandler`, but not a change to the `Handler` interface, since there is no conceivable reason any other handler would implement it.

Another option is to provide a `handler.shouldSkipReportingOnInitializer(...)` or something slightly more generic, if we think this might be a recurring need for other methods/handlers in the future.

I still think the best approach is to skip this whole thing and switch to using a shadow field. Although it is possible that I am missing (or have forgotten) something.

cc: @msridhar @navahgar